gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -582,11 +581,18 @@ private void cleanUp() {
       }
     }
 
-    private void cleanUpDag(String dagId) {
-      Dag<JobExecutionPlan> dag = this.dags.get(dagId);
-      this.dagToJobs.remove(dagId);
-      this.dagStateStore.cleanUp(dag);
+    /**
+     * Cleaning of all relevant states need to be
+     * @param dagId
+     */
+    private synchronized void cleanUpDag(String dagId) {
+       try {
+         this.dagStateStore.cleanUp(dags.get(dagId));
+       } catch (IOException ioe) {
+         log.error(String.format("Failed to clean %s from backStore due to:", dagId), ioe);
```

Comment:
log.error("Failed to clean {} from backStore due to: ", dagId, ioe);?

Reply:
I believe `log.error (String, Exception)` will format the line better. What is the reason to have them cancatted as single string ?

Reply:
Not necessarily better. I have seen parametrized logging being used more frequently in Gobblin code.